### PR TITLE
Add GitHub actions

### LINKED ISSUE #59: Setup continuous deployments (releases + ghcr.io docker image)
Let's write some GitHub Actions to automatically 
- [x] run the unit tests (on every push)
  - [x] Core library
  - [x] Demo
- [x] deploy releases (on every merge into `main`)
  - [x] bump version
  - [x] create and push git tag with new version
  - [x] deploy tar as release asset
  - [x] build & deploy docker image with latest tag to ghcr.io
    - [x] [Core library](https://github.com/kit-mrt/arbitration_graphs/pkgs/container/arbitration_graphs) 
    - [x] [Demo](https://github.com/kit-mrt/arbitration_graphs/pkgs/container/arbitration_graphs_pacman_demo)
    - [x] [Tutorial](https://github.com/kit-mrt/arbitration_graphs/pkgs/container/arbitration_graphs_pacman_tutorial)
  - [x] update `latest` tags of docker images on ghcr.io

We should make sure to also
- [x] cleanup and update the main Readme w.r.t. installation options

### PR DESCRIPTION
**Pull request by ll-nick**
See #59  for details.

This adds automated workflows to 
- build and run unit tests of both the core library and the demo on any push to any branch
- on merge to main:
  - bump the version in the `version` file according to the version string in the commit messages (`#major`, `#minor`, `#patch`, `#none`). `#minor` by default.
  -  create a new release and add debian packages as well as tar.gz and zip release
  - run the unit tests against the released packages
  - build and push new docker images for the core library, the demo and the tutorial

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -47,4 +47,24 @@ RUN mkdir /tmp/arbitration_graphs/build && \
...
+COPY version /tmp/arbitration_graphs/version
+
+# Prepare build directory
+RUN mkdir /tmp/arbitration_graphs/build && \
+    cd /tmp/arbitration_graphs/build && \
+    cmake -DBUILD_TESTS=true .. && \
+    cmake --build . -j9
+
+# Run unit tests
+CMD ["cmake", "--build", "/tmp/arbitration_graphs/build", "--target", "test"]
```
Comment (orzechow) [inline Dockerfile:69]: :ok_hand:

**Comment by ll-nick** [comment]
Alright @orzechow , this one is ready for review. It should be very analogous to the util_caching pipelines now. Feel free to push your release-test-pipeline onto this branch if we want to do that in one go.

**Comment by orzechow** [comment]
> Alright @orzechow , this one is ready for review. It should be very analogous to the util_caching pipelines now. Feel free to push your release-test-pipeline onto this branch if we want to do that in one go.

Thanks!

One quick remark: the dockerfile is missing the release_builder stage etc.

Yeah, I'll add the release test pipeline after finishing the cpack PR

**Comment by ll-nick** [comment]
You're right. I think it's easier if one person just does it. I have moved the PR back to draft to wait until you finished the cpack PR.

I will then rebase this PR, add the Docker stage and release test pipeline and ping you once the PR is actually done.

**Comment by ll-nick** [comment]
Rebased on main

**Comment by ll-nick** [comment]
Ok @orzechow this one should truly be ready now.

See the edited PR description for all the things that are now going on the workflows.

I tested the workflow on a [test branch](https://github.com/KIT-MRT/arbitration_graphs/tree/test_github_actions) where I didn't need a PR to trigger the workflow, see 82b31a0d2e551b35fb531e5b0fb21f45be565fc6. Other than that, the branch is identical to this one and the [last run was succesful](https://github.com/KIT-MRT/arbitration_graphs/actions/runs/11837536269).

A couple of noteworthy changes I made since we last talked about this:
- Combine the release and the docker build workflows. This is due to the fact that we need the version bump to happen first. Otherwise the docker images will be tagged with the old version. Also I guess it's nice to wait for the release tests before publishing new images.
- The unit-test workflow now build the library core image before building the demo unit test stage. This is due to the fact that we want to test the state that the repo is in. Otherwise, the demo unit tests would run against the version that's currently hosted on ghcr (which doesn't even necessarily exist).

**Pull request merged**